workflow/controller/operator.go

```diff
@@ -2248,9 +2248,9 @@ func (woc *wfOperationCtx) executeTemplate(ctx context.Context, nodeName string,
 	}
 
 	if processedTmpl.Metrics != nil {
-		// Check if the node was just created, if it was emit realtime metrics.
+		// Check if the node was just created or not fulfilled, if it was emit realtime metrics.
 		// If the node did not previously exist, we can infer that it was created during the current operation, emit real time metrics.
-		if _, ok := woc.preExecutionNodePhases[node.ID]; !ok {
+		if _, ok := woc.preExecutionNodePhases[node.ID]; !ok || !node.Fulfilled() {
```

Comment:
@Joibel What I want to modify is at the template level, my reason is that I specified `when duration > 100`, which means that the **realtime** metric is actually sent out when it is greater than 100 seconds. In fact, this metric will not be collected when it is greater than 100 seconds, because the duration is 0 when the template is just created. When calculated as false, I permanently lost this metrics.

Reply:
I think this does not conform to the definition of "when".

Reply:
My expectation is that I can collect this metrics when the `when condition` is met.

Reply:
Could you read my suggestion again. I believe it does what you're looking for here.

Reply:
> Could you read my suggestion again. I believe it does what you're looking for here.

I understand, thank you.